content/java/concepts/hash-map/hash-map.md

```diff
@@ -0,0 +1,216 @@
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'HashMap'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+In Java, a HashMap is used to store items as a "key/value" pairs.
+
+The keys and values can be of either same or different types.
+
+```java
+// Create a HashMap which stores courses and their teachers as a key/value pair 
+HashMap<String, String> courseTeacher = new HashMap<String, String>();
+```
+
+## Creating a HashMap
+
+In Java, a `HashMap` can be created in the following way:
+
+Identify the datatype of the key and the value for the key-value pair. Then follow the following template to create a new `HashMap`.
+
+`HashMap<KeyDatatype, ValueDatatype> HashMapName = new HashMap<KeyDatatype, ValueDatatype>();`
+
+```java
+  import java.util.HashMap; // import the HashMap class
+
+  // Create a HashMap which stores courses and their teachers as a key/value pair 
+  HashMap<String, String> courseTeacher = new HashMap<String, String>();
+```
+
+Here, courseTeacher stores the key value Course which is of `String` type and the value are the Teacher assigned for the course which is also of `String` type.
+
+```java
+  import java.util.HashMap; // import the HashMap class
+
+  // Create a HashMap which stores students and their marks as a key/value pair
+  HashMap<String, Integer> studentMarks = new HashMap<String, Integer>();
+```
+
+Here, studentMarks stores the key value Student name which is of `String` type and the value are the Marks secured by the student which is also of `Integer` type.
+
+## Key-Value
+
+A `Key` can be used to uniquely identify a `Value` in a `HashMap`. This allows for efficient data storage and easy access. In the example below the `Course Names` are the `Keys` and the `Teacher` assigned are the `Values` which can be accessed using the `Key` which is the corresponding `Course Names`.
+
+```java
+import java.util.HashMap; // import the HashMap class
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // Add keys and values (CourseNames, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    System.out.println(courseTeacher.get("Physics"););
+    // Output: Lily
+
+    System.out.println(courseTeacher.get("History"););
+    // Output: Ben
+  }
+}
+```
+
+## Adding Items
+
+In Java, an item can be added to a `HashMap` using the `put()` method.
+
+The `put()` method takes in 2 attributes. The first attribute is the key and the second attribute is the corresponding value.
+
+```java
+import java.util.HashMap; // import the HashMap class
```

Comment:
```suggestion
// Import the HashMap class
import java.util.HashMap;
```

Although I'm suggesting this for each import statement, one thing we can do is add a line about it when we brought it into our code snippet for the first time. If there is an explanation or even a **Note:** (outside not inside the code snippet), we could even omit the import statements that follow the one time it says `import java.util.HashMap;`.